In [1]:
# Importing the libraries
import pymysql
import pandas as pd

In [2]:
# Database Credentials
db_name = "sales"
db_host = "localhost" # default
db_username = "root"   # default
db_password = "password"

In [3]:
# Establish the connection
try:
    conn = pymysql.connect(host= db_host,
                           port= int(3306),
                           user = 'root',
                           passwd=db_password,
                           db=db_name)
    
except e:
    print(e)

In [4]:
# Test the connection to the database
if conn:
    print(" Successfully connected")
else:
    print("Error")

 Successfully connected


#### 1. How many customers does the company have?

In [43]:
df1 = pd.read_sql_query("SELECT DISTINCT COUNT(customer_code) AS TotalNumberofCustomers FROM sales.customers", conn)

C:\Users\LENOVO\anaconda3\lib\site-packages\pandas\io\sql.py:761: UserWarning: pandas only support SQLAlchemy connectable(engine/connection) ordatabase string URI or sqlite3 DBAPI2 connectionother DBAPI2 objects are not tested, please consider using SQLAlchemy
  warnings.warn(


In [44]:
df1

TotalNumberofCustomers
0                      38

There are total 38 customers across different regions for this company

#### 2. What is the distribution type of the customers and their respective revenue generated?

In [55]:
query1 = """

SELECT customer_type , SUM(sales_amount) AS total_revenue
FROM sales.transactions as t
LEFT JOIN sales.customers as c
ON t.customer_code = c.customer_code
GROUP BY customer_type
       
       
       """

In [56]:
df2 = pd.read_sql_query(query1, conn)

C:\Users\LENOVO\anaconda3\lib\site-packages\pandas\io\sql.py:761: UserWarning: pandas only support SQLAlchemy connectable(engine/connection) ordatabase string URI or sqlite3 DBAPI2 connectionother DBAPI2 objects are not tested, please consider using SQLAlchemy
  warnings.warn(


In [57]:
df2

customer_type  total_revenue
0  Brick & Mortar    746198243.0
1      E-Commerce    240367523.0

Customer who own a Brick and Mortar have purchased more products from the company indicating that offline sales channels have been more profitable for the company than the online sales through e-commerce webistes.

#### 3. What is the sales revenue trend of the Company?

In [58]:
query2 = """

SELECT d.year, SUM(sales_amount) AS total_revenue
FROM sales.transactions as t
INNER JOIN sales.date as d
ON t.order_date = d.date
GROUP BY d.year
       
       
"""

In [59]:
df3 = pd.read_sql_query(query2, conn)

C:\Users\LENOVO\anaconda3\lib\site-packages\pandas\io\sql.py:761: UserWarning: pandas only support SQLAlchemy connectable(engine/connection) ordatabase string URI or sqlite3 DBAPI2 connectionother DBAPI2 objects are not tested, please consider using SQLAlchemy
  warnings.warn(


In [60]:
df3

year  total_revenue
0  2017     93569152.0
1  2018    414308941.0
2  2019    336452114.0
3  2020    142235559.0

The revenue for the year 2018 has been the highest compared to all the other years. We can also note that there has been a steep decline in the revenue through the years 2018 to 2020.

#### 4. Which month has shown the highest Sales order Volume?

In [84]:
query3 = """

SELECT d.month_name AS MonthName, COUNT(sales_amount) AS OrderSize
FROM sales.transactions as t
INNER JOIN sales.date as d
ON t.order_date = d.date
GROUP BY d.month_name
ORDER BY OrderSize DESC
       
       
"""

In [85]:
df4 = pd.read_sql_query(query3, conn)

C:\Users\LENOVO\anaconda3\lib\site-packages\pandas\io\sql.py:761: UserWarning: pandas only support SQLAlchemy connectable(engine/connection) ordatabase string URI or sqlite3 DBAPI2 connectionother DBAPI2 objects are not tested, please consider using SQLAlchemy
  warnings.warn(


In [86]:
df4

MonthName  OrderSize
0    November      14351
1     January      14014
2     October      13763
3    February      13685
4         May      13584
5       March      13485
6       April      13188
7    December      12699
8        June      12666
9        July      10363
10     August       9470
11  September       9015

An increase in the order placed occured in the month of November and January followed by October.

#### 5. Which zone has grossed the highest revenue for the company?

In [79]:
query4 = """

SELECT zone AS Zone, SUM(sales_qty) as OrderVolume
FROM sales.transactions as t
INNER JOIN sales.markets as m
ON t.market_code = m.markets_code
GROUP BY zone
ORDER BY OrderVolume DESC
       
       
"""

In [80]:
df5 = pd.read_sql_query(query4, conn)

C:\Users\LENOVO\anaconda3\lib\site-packages\pandas\io\sql.py:761: UserWarning: pandas only support SQLAlchemy connectable(engine/connection) ordatabase string URI or sqlite3 DBAPI2 connectionother DBAPI2 objects are not tested, please consider using SQLAlchemy
  warnings.warn(


In [81]:
df5

Zone  OrderVolume
0    North    1276395.0
1  Central     764519.0
2    South     403501.0

North Zone has generated the highest revenue followed by the Central and South Zone respectively.

#### 6. Which individual markets in the North had the highest Sales Order Volume?

In [78]:
query5 = """

SELECT zone, markets_name AS MarketName, SUM(sales_qty) as SalesQty
FROM sales.transactions as t
INNER JOIN sales.markets as m
ON t.market_code = m.markets_code
WHERE zone = 'North'
GROUP BY zone, markets_name
ORDER BY SalesQty DESC
       
       
"""

In [82]:
df6 = pd.read_sql_query(query5, conn)

C:\Users\LENOVO\anaconda3\lib\site-packages\pandas\io\sql.py:761: UserWarning: pandas only support SQLAlchemy connectable(engine/connection) ordatabase string URI or sqlite3 DBAPI2 connectionother DBAPI2 objects are not tested, please consider using SQLAlchemy
  warnings.warn(


In [83]:
df6

zone MarketName  SalesQty
0  North  Delhi NCR  992338.0
1  North  Ahmedabad  207575.0
2  North    Lucknow   37092.0
3  North      Surat   17200.0
4  North     Kanpur   16677.0
5  North      Patna    5513.0

Delhi NCR has the generated the highest Order size as compared to the markets in other North region such as Surat, Kanpur, Patna etc.

#### 7. Minimum and Maximum Order Size for Sales Revenue greater than Rs 10k

In [75]:
query6 = """

SELECT MAX(sales_qty) AS MaxOrderSize, MIN(sales_qty) AS MinOrderSize
FROM sales.transactions as t
WHERE sales_qty > 0 and sales_amount > 10000
       
       
"""

In [76]:
df6 = pd.read_sql_query(query6, conn)

C:\Users\LENOVO\anaconda3\lib\site-packages\pandas\io\sql.py:761: UserWarning: pandas only support SQLAlchemy connectable(engine/connection) ordatabase string URI or sqlite3 DBAPI2 connectionother DBAPI2 objects are not tested, please consider using SQLAlchemy
  warnings.warn(


In [77]:
df6

MaxOrderSize  MinOrderSize
0         14049             3

The maximum order size for a customer is 14049 while the minimum order size is 3.